shm/inc/player.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <memory>
+
+#include "ship.hpp"
+
+class Player {
+public:
+    Player(Ship& ship, size_t money, size_t availableSpace);
+
+    std::unique_ptr<Ship> getShip() { return std::move(ship_); }
```

Comment:
You can use this method only once, because you return ship_ and you will loose your handler to this pointer.

Reply:
Will std::shared_ptr<> do the job here?

Reply:
getShip() should return Ship*. And you will return ship_.get(). If you don't want to make any changes in ship, only acces to const method you can return as const ptr. 

When you use shared_ptr you can return shared_ptr<Ship>